### PR TITLE
Archive and Update project command

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
# Archive and Update project command
Usage
1] Added new command named update
Test Output:
```
bin/flytectl --help
Usage:
   [command]

Available Commands:
  config      Runs various config commands, look at the help of this command to get a list of available commands..
  get         Retrieve various resource.
  help        Help about any command
  update      Update various resources.
  version     Displays version information for the client and server.
```
2] Added project as available command under update and flags for activate and archive
```
bin/flytectl update project --help
Retrieves project resources.

Usage:
   update project [flags]

Aliases:
  project, projects

Flags:
  -t, --activateProject   Activates the project specified as argument.
  -a, --archiveProject    Archives the project specified as argument.
  -h, --help              help for project
```
3] No project Input

```
bin/flytectl update project
INFO[0000] Using config file: [config.yaml]             
Project  not found
```
4] Without update options
```
 bin/flytectl update project -p flytesnacks
INFO[0000] Using config file: [config.yaml]             
Invalid state passed. Specify either activate or archive
```
5] Archive a project

```
bin/flytectl update project -p flytesnacks --archive
INFO[0000] Using config file: [config.yaml]             
INFO[0000] Config section [root] updated. No update handler registered.  src="viper.go:318"
INFO[0000] Config section [logger] updated. Firing updated event.  src="viper.go:320"
{"json":{"src":"viper.go:320"},"level":"info","msg":"Config section [admin] updated. Firing updated event.","ts":"2021-01-29T17:17:25+05:30"}
{"json":{"src":"viper.go:318"},"level":"info","msg":"Config section [update] updated. No update handler registered.","ts":"2021-01-29T17:17:25+05:30"}
{"json":{"src":"viper.go:318"},"level":"info","msg":"Config section [adminutils] updated. No update handler registered.","ts":"2021-01-29T17:17:25+05:30"}
{"json":{"src":"client.go:29"},"level":"info","msg":"Initialized Admin client","ts":"2021-01-29T17:17:25+05:30"}
Project flytesnacks updated to ARCHIVED state
```
6] Output after archival from get command

```
bin/flytectl  get project 
| ID            | NAME          | DESCRIPTION               |
 --------------- --------------- --------------------------- 
| flyteexamples | flyteexamples | flyteexamples description |
| flytetester   | flytetester   | flytetester description   |
```
7] Activate project

```
bin/flytectl update project -p flytesnacks --activateProject
INFO[0000] Using config file: [config.yaml]             
INFO[0000] Config section [adminutils] updated. No update handler registered.  src="viper.go:318"
INFO[0000] Config section [root] updated. No update handler registered.  src="viper.go:318"
INFO[0000] Config section [logger] updated. Firing updated event.  src="viper.go:320"
{"json":{"src":"viper.go:320"},"level":"info","msg":"Config section [admin] updated. Firing updated event.","ts":"2021-01-29T17:18:10+05:30"}
{"json":{"src":"viper.go:318"},"level":"info","msg":"Config section [update] updated. No update handler registered.","ts":"2021-01-29T17:18:10+05:30"}
{"json":{"src":"client.go:29"},"level":"info","msg":"Initialized Admin client","ts":"2021-01-29T17:18:10+05:30"}
Project flytesnacks updated to ACTIVE state
```

8] Output post activation

```
bin/flytectl  get project 
| ID            | NAME          | DESCRIPTION               |
 --------------- --------------- --------------------------- 
| flyteexamples | flyteexamples | flyteexamples description |
| flytesnacks   | flytesnacks   | flytesnacks description   |
| flytetester   | flytetester   | flytetester description   |
3 rows
```
9] Project not found

```
bin/flytectl update project -p invalidproject --activate
Project invalidproject failed to get updated to ACTIVE state due to rpc error: code = NotFound desc = entry not found
```

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
Not found. Will create one.
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: cmd/update/update.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+		Use:   "update",
+		Short: "Update various resources.",
+	}
+
+	updateResourcesFuncs := map[string]cmdcore.CommandEntry{
+		"project":    {CmdFunc: updateProjectsFunc, Aliases: []string{"projects"}, ProjectDomainNotRequired: true},
+	}
+
+	cmdcore.AddCommands(updateCmd, updateResourcesFuncs)
+	updateCmd.PersistentFlags().BoolVarP(&(GetConfig().ActivateProject), "activate", "t", false, "Activates the project specified as argument.")
```
Comment (kumare3) [inline cmd/update/update.go:21]: These flags should be only for the `project` command. Currently you have added them to all update commands, which is not correct

File: cmd/update/config.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+var (
+	defaultConfig = &Config{
+	}
+	section       = config.MustRegisterSection("update", defaultConfig)
+)
+
+// Config hold configuration for project update flags.
+type Config struct {
+	ActivateProject bool `json:"activateProject" pflag:",Activates the project specified as argument."`
```
Comment (kumare3) [inline cmd/update/config.go:17]: Can you also add a small section to the contributing guide, that all `flags` are camel cased

File: cmd/update/update.go (hunk 1 above)
Comment (kumare3) [reply cmd/update/update.go:21]: this has to be changed

**Comment by kumare3** [comment]
Once you rebase on #22 +1 from me

File: cmd/update/update.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply cmd/update/update.go]: Was waiting on my other MR to be merged. Took care of this.

File: cmd/update/config.go (hunk 2 above)
Comment (pmahindrakar-oss) [reply cmd/update/config.go]: Done

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -4,12 +4,13 @@ A local cluster can be setup via --> https://lyft.github.io/flyte/administrator/
 
 Then, if having trouble connecting to local cluster see the following:
 
+
 #1) Find/Set/Verify gRPC port for your local Flyte service:
 
 FLYTECTL_GRPC_PORT=`kubectl get service -n flyte flyteadmin -o json | jq '.spec.ports[] | select(.name=="grpc").port'`
```
Comment (kumare3) [inline docs/CONTRIBUTING.md:9]: This just needs the envoy ingress now.

**Review by kumare3 — APPROVED** [review]

**Pull request merged**